server/src/services/postgrest_proxy.service.ts

```diff
@@ -6,63 +6,61 @@ import * as proxy from 'express-http-proxy';
+    req.url = await this.replaceTableNamesAtPlaceholder(req.url, organizationId);
     const authToken = 'Bearer ' + this.signJwtPayload(this.configService.get<string>('PG_USER'));
     req.headers = {};
     req.headers['Authorization'] = authToken;
     req.headers['Prefer'] = 'count=exact'; // To get the total count of records
 
     res.set('Access-Control-Expose-Headers', 'Content-Range');
 
-    const tableId = req.url.split('?')[0].split('/').pop();
-    const internalTable = await this.manager.findOne(InternalTable, {
-      where: {
-        organizationId,
-        id: tableId,
-      },
-    });
-
-    if (internalTable.tableName) {
-      const tableInfo = {};
-      tableInfo[tableId] = internalTable.tableName;
-      req.headers['tableInfo'] = tableInfo;
-    }
-
     return this.httpProxy(req, res, next);
   }
 
-  private httpProxy = proxy(this.configService.get<string>('PGRST_HOST') || 'http://localhost:3001', {
-    userResDecorator: function (proxyRes, proxyResData, userReq, _userRes) {
-      if (userReq?.headers?.tableInfo && proxyRes.statusCode >= 400) {
-        const customErrorObj = Buffer.isBuffer(proxyResData) ? JSON.parse(proxyResData.toString('utf8')) : proxyResData;
-
-        let customErrorMessage = customErrorObj?.message ?? '';
-        if (customErrorMessage) {
-          Object.entries(userReq.headers.tableInfo).forEach(([key, value]) => {
-            customErrorMessage = customErrorMessage.replace(key, value as string);
-          });
-          customErrorObj.message = customErrorMessage;
-        }
-        proxyResData = Buffer.from(JSON.stringify(customErrorObj), 'utf-8');
+  async perform(url, method, headers, body) {
+    try {
+      const authToken = 'Bearer ' + this.signJwtPayload(this.configService.get<string>('PG_USER'));
+      const updatedPath = replaceUrlForPostgrest(url);
+      let postgrestUrl = (this.configService.get<string>('PGRST_HOST') || 'http://localhost:3001') + updatedPath;
+
+      if (!postgrestUrl.startsWith('http://') && !postgrestUrl.startsWith('https://')) {
+        postgrestUrl = 'http://' + postgrestUrl;
       }
 
-      return proxyResData;
-    },
-    proxyReqPathResolver: function (req) {
-      const path = '/api/tooljet-db';
-      const pathRegex = new RegExp(`${maybeSetSubPath(path)}/proxy`);
-      const parts = req.url.split('?');
-      const queryString = parts[1];
-      const updatedPath = parts[0].replace(pathRegex, '');
+      const reqHeaders = {
+        ...headers,
+        Authorization: authToken,
+        Prefer: 'count=exact', // get the total no of records
+      };
+
+      const response = await got(postgrestUrl, {
+        method,
+        headers: reqHeaders,
+        responseType: 'json',
+        throwHttpErrors: false, // don't throw for non-2xx status codes
+        ...(!isEmpty(body) && { body: JSON.stringify(body) }),
+      });
+
+      return response.body;
+    } catch (error) {
+      return error;
```

Comment:
Shouldn't we throw error, to understand/display error message on front end? or do we have any other ways to capture error

Reply:
You are right, we should throw `QueryError` here instead as this will be invoked from query execution.